cpp/arcticdb/storage/s3/real_s3_client.hpp

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#pragma once
+
+#include <aws/s3/S3Client.h>
+
+#include <arcticdb/storage/s3/s3_client_wrapper.hpp>
+
+#include <arcticdb/util/preconditions.hpp>
+#include <arcticdb/util/pb_util.hpp>
+#include <arcticdb/log/log.hpp>
+#include <arcticdb/util/buffer_pool.hpp>
+
+#include <arcticdb/storage/object_store_utils.hpp>
+#include <arcticdb/storage/storage_utils.hpp>
+#include <arcticdb/entity/serialized_key.hpp>
+#include <arcticdb/util/exponential_backoff.hpp>
+#include <arcticdb/util/configs_map.hpp>
+#include <arcticdb/util/composite.hpp>
+
+namespace arcticdb::storage::s3{
+
+// A real S3ClientWrapper around Aws::S3::Client, which executes actual requests to S3 storage.
+class RealS3Client : public S3ClientWrapper {
+public:
+    template<typename ...Args>
+    RealS3Client(Args && ...args):s3_client(std::forward<Args>(args)...){};
```

Comment:
Can't we just have a `using S3ClientWrapper::S3ClientWrapper` to get the base constructors?